operations/host_spawn.go

```diff
@@ -221,6 +223,93 @@ func hostModify() cli.Command {
 	}
 }
 
+func hostConfigure() cli.Command {
+	return cli.Command{
+		Name:  "configure",
+		Usage: "run setup commands for a virtual workstation",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  joinFlagNames(projectFlagName, "p"),
+				Usage: "name of an evergreen project to run commands from",
+			},
+			cli.StringFlag{
+				Name:  joinFlagNames(dirFlagName, "d"),
+				Usage: "directory to run commands from (will override project configuration)",
+			},
+			cli.BoolFlag{
+				Name:  joinFlagNames(quietFlagName, "q"),
+				Usage: "suppress output",
+			},
+		},
+		Before: mergeBeforeFuncs(setPlainLogger, requireProjectFlag),
+		Action: func(c *cli.Context) error {
+			confPath := c.Parent().Parent().String(confFlagName)
+			project := c.String(projectFlagName)
+			directory := c.String(dirFlagName)
+			quiet := c.Bool(quietFlagName)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			ac, _, err := conf.getLegacyClients()
+			if err != nil {
+				return errors.Wrap(err, "problem accessing evergreen service")
+			}
+
+			projectRef, err := ac.GetProjectRef(project)
+			if err != nil {
+				return errors.Wrapf(err, "can't find project for queue id '%s'", project)
+			}
+			cmds, err := getJasperCommands(projectRef, directory, quiet)
+			if err != nil {
+				return errors.Wrapf(err, "error getting commands")
+			}
+			for _, cmd := range cmds {
+				if err := cmd.Run(ctx); err != nil {
+					gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "error running command"))
+				}
+			}
+			return nil
+		},
+	}
+}
+
+func getJasperCommands(projectRef *model.ProjectRef, directory string, quiet bool) ([]*jasper.Command, error) {
+	cmds := []*jasper.Command{}
+	userHome, err := homedir.Dir()
+	if err != nil {
+		return nil, errors.Wrap(err, "problem finding home directory")
+	}
+	if len(projectRef.WorkstationConfig) == 0 {
+		return nil, errors.Errorf("no commands configured for project '%s'", projectRef.Identifier)
+	}
+	for _, obj := range projectRef.WorkstationConfig {
+		dir := userHome
+		if directory != "" {
+			dir = fmt.Sprintf("%s/%s", dir, directory)
+		} else if obj.Directory != "" {
+			dir = fmt.Sprintf("%s/%s", dir, obj.Directory)
+		}
+		// what if a user wants to clone something that isn't this default? Should the command EQUAL git clone?
+		if strings.Contains(obj.Command, "git clone") {
+			// add repo/branch
+			obj.Command = fmt.Sprintf("git clone -b %s git@github.com:%s/%s.git", projectRef.Branch, projectRef.Owner, projectRef.Repo)
```

Comment:
I worry about git cloning vs https cloning? this is likely to fail if people don't have SSH keys setup correctly. should we give people an option?